### ISSUE TITLE
FastUpToDate check seems to look for content in the wrong output folder (Perf.Tests in Roslyn)

### ISSUE THREAD

**Issue by jcouv** [dotnet/project-system#2842]
I noticed a problem with building Roslyn solution (recent master branch). The FastUpToDate check fails on some of them (20 out of 145), when it should succeed.
For example, the `Perf.Tests` project (`src\Test\Perf\tests\Perf.Tests.csproj`).

I delete the output folder (`D:\repos\roslyn3\Binaries\Debug\Dlls\Perf.Tests`), re-build, then build, then build again. I expect the FastUpToDate check to succeed in that last build. But the check fails (see log below).

This project is just copying three content files, which are in a small folder structure.
For some reason, the FastUpToDate check seems to look for the file in the wrong output location (it should be in a sub-folder)

1>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\Dlls\Perf.Tests\ **missing sub-folder here** HelloWorld.cs' does not exist. (Perf.Tests)

Re-build:
```
1>------ Rebuild All started: Project: Perf.Tests, Configuration: Debug Any CPU ------
1>Perf.Tests -> D:\repos\roslyn3\Binaries\Debug\Dlls\Perf.Tests\Roslyn.Test.Performance.Tests.dll
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========
```

Then Build:
```
1>FastUpToDate: Latest write timestamp on input is 9/19/2017 5:08:16 PM on 'D:\repos\roslyn3\src\Test\Perf\tests\Perf.Tests.csproj'. (Perf.Tests)
1>FastUpToDate: Earliest write timestamp on output is 9/19/2017 5:12:03 PM on 'D:\repos\roslyn3\Binaries\Obj\Perf.Tests\Debug\Roslyn.Test.Performance.Tests.xml'. (Perf.Tests)
1>FastUpToDate: Checking PreserveNewest file 'D:\repos\roslyn3\src\Test\Perf\tests\helloworld\HelloWorld.cs': (Perf.Tests)
1>FastUpToDate:     Write 7/14/2017 10:09:29 PM: 'D:\repos\roslyn3\src\Test\Perf\tests\helloworld\HelloWorld.cs'. (Perf.Tests)
1>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\Dlls\Perf.Tests\HelloWorld.cs' does not exist. (Perf.Tests)
1>FastUpToDate: Project is not up to date. (Perf.Tests)
1>------ Build started: Project: Perf.Tests, Configuration: Debug Any CPU ------
1>Perf.Tests -> D:\repos\roslyn3\Binaries\Debug\Dlls\Perf.Tests\Roslyn.Test.Performance.Tests.dll
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

FYI @panopticoncentral @davkean 

Here are some other instances of what seems to be the same problem (the content file exists in the output, but it is in a sub-folder):
```

92>FastUpToDate: Checking PreserveNewest file 'D:\repos\roslyn3\src\EditorFeatures\Test\PerfTests\ResetRoslynOptions.xml': (EditorServicesTest)
92>FastUpToDate:     Write 2/26/2016 11:01:47 PM: 'D:\repos\roslyn3\src\EditorFeatures\Test\PerfTests\ResetRoslynOptions.xml'. (EditorServicesTest)
92>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\EditorServicesTest\ResetRoslynOptions.xml' does not exist. (EditorServicesTest)
92>FastUpToDate: Project is not up to date. (EditorServicesTest)
92>------ Build started: Project: EditorServicesTest, Configuration: Debug Any CPU ------

99>FastUpToDate: Checking PreserveNewest file 'D:\repos\roslyn3\src\EditorFeatures\CSharpTest\PerfTests\Sources\CSharpPgoTypingInput.txt': (CSharpEditorServicesTest)
99>FastUpToDate:     Write 2/26/2016 11:01:46 PM: 'D:\repos\roslyn3\src\EditorFeatures\CSharpTest\PerfTests\Sources\CSharpPgoTypingInput.txt'. (CSharpEditorServicesTest)
99>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\CSharpEditorServicesTest\CSharpPgoTypingInput.txt' does not exist. (CSharpEditorServicesTest)
99>FastUpToDate: Project is not up to date. (CSharpEditorServicesTest)
99>------ Build started: Project: CSharpEditorServicesTest, Configuration: Debug Any CPU ------

110>FastUpToDate: Checking PreserveNewest file 'D:\repos\roslyn3\src\ExpressionEvaluator\CSharp\Test\ExpressionCompiler\PerfTests\DebuggerSteppingTestcase.xml': (CSharpExpressionCompilerTest)
110>FastUpToDate:     Write 2/26/2016 11:01:48 PM: 'D:\repos\roslyn3\src\ExpressionEvaluator\CSharp\Test\ExpressionCompiler\PerfTests\DebuggerSteppingTestcase.xml'. (CSharpExpressionCompilerTest)
110>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\CSharpExpressionCompilerTest\DebuggerSteppingTestcase.xml' does not exist. (CSharpExpressionCompilerTest)
110>FastUpToDate: Project is not up to date. (CSharpExpressionCompilerTest)
110>------ Build started: Project: CSharpExpressionCompilerTest, Configuration: Debug Any CPU ------

114>FastUpToDate: Checking PreserveNewest file 'D:\repos\roslyn3\src\EditorFeatures\VisualBasicTest\PerfTests\Sources\BasicPgoTypingInput.txt': (BasicEditorServicesTest)
114>FastUpToDate:     Write 2/26/2016 11:01:48 PM: 'D:\repos\roslyn3\src\EditorFeatures\VisualBasicTest\PerfTests\Sources\BasicPgoTypingInput.txt'. (BasicEditorServicesTest)
114>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\BasicEditorServicesTest\BasicPgoTypingInput.txt' does not exist. (BasicEditorServicesTest)
114>FastUpToDate: Project is not up to date. (BasicEditorServicesTest)
114>------ Build started: Project: BasicEditorServicesTest, Configuration: Debug Any CPU ------
```

Looking at the other Roslyn projects where FastUpToDate check fail, I suspect they are also due to the same problem:
```

80>FastUpToDate: Checking PreserveNewest file 'C:\Users\jcouv\.nuget\packages\netstandard.library\2.0.0-beta-25123-01\build\netstandard2.0\ref\mscorlib.dll': (CSharpCompilerSymbolTest)
80>FastUpToDate:     Write 3/23/2017 1:23:56 PM: 'C:\Users\jcouv\.nuget\packages\netstandard.library\2.0.0-beta-25123-01\build\netstandard2.0\ref\mscorlib.dll'. (CSharpCompilerSymbolTest)
80>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\CSharpCompilerSymbolTest\netcoreapp2.0\mscorlib.dll' does not exist. (CSharpCompilerSymbolTest)
80>FastUpToDate: Project is not up to date. (CSharpCompilerSymbolTest)
80>------ Build started: Project: CSharpCompilerSymbolTest, Configuration: Debug Any CPU ------

81>FastUpToDate: Checking PreserveNewest file 'C:\Users\jcouv\.nuget\packages\netstandard.library\2.0.0-beta-25123-01\build\netstandard2.0\ref\mscorlib.dll': (CSharpCompilerSyntaxTest)
81>FastUpToDate:     Write 3/23/2017 1:23:56 PM: 'C:\Users\jcouv\.nuget\packages\netstandard.library\2.0.0-beta-25123-01\build\netstandard2.0\ref\mscorlib.dll'. (CSharpCompilerSyntaxTest)
81>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\UnitTests\CSharpCompilerSyntaxTest\netcoreapp2.0\mscorlib.dll' does not exist. (CSharpCompilerSyntaxTest)
81>FastUpToDate: Project is not up to date. (CSharpCompilerSyntaxTest)
81>------ Build started: Project: CSharpCompilerSyntaxTest, Configuration: Debug Any CPU ------

84>FastUpToDate:     Write 1/4/2017 5:55:08 PM: 'C:\Users\jcouv\.nuget\packages\sqlitepclraw.lib.e_sqlite3.v110_xp\1.1.2\runtimes\win7-x86\native\e_sqlite3.dll'. (VisualStudioSetup)
84>FastUpToDate:     Output file 'D:\repos\roslyn3\Binaries\Debug\Vsix\VisualStudioSetup\e_sqlite3.dll' does not exist. (VisualStudioSetup)
84>FastUpToDate: Project is not up to date. (VisualStudioSetup)
84>------ Build started: Project: VisualStudioSetup, Configuration: Debug Any CPU ------
```

**Comment by jcouv** [comment]
FYI @TyOverby

**Comment by panopticoncentral** [comment]
I will try and take a look at this soon.

**Comment by davkean** [comment]
@panopticoncentral Any progress on this?

**Comment by gulbanana** [comment]
i'm also experiencing this problem; it breaks all test projects, since xunit copies some files to the output.
feels like a regression of https://github.com/dotnet/project-system/issues/2717

**Comment by panopticoncentral** [comment]
With the fix to #3023, all the projects will eventually be up to date with the exception of projects with app.config files, which is covered by #2695.
